test-framework/test_runner.go

```diff
@@ -0,0 +1,474 @@
+			program := interpreter.ProgramFromChecker(stdlib.TestContractChecker)
+			subInterpreter, err := inter.NewSubInterpreter(program, location)
+			if err != nil {
+				panic(err)
+			}
+			return interpreter.InterpreterImport{
+				Interpreter: subInterpreter,
+			}
+
+		default:
+			importedProgram, importedElaboration, err := r.parseAndCheckImport(location, ctx)
+			if err != nil {
+				panic(err)
+			}
+
+			program := &interpreter.Program{
+				Program:     importedProgram,
+				Elaboration: importedElaboration,
+			}
+
+			subInterpreter, err := inter.NewSubInterpreter(program, location)
+			if err != nil {
+				panic(err)
+			}
+
+			return interpreter.InterpreterImport{
+				Interpreter: subInterpreter,
+			}
+		}
+	}
+}
+
+// newScriptEnvironment creates an environment for test scripts to run.
+// Leverages the functionality of FVM.
+//
+func newScriptEnvironment() *fvm.ScriptEnv {
+	vm := fvm.NewVirtualMachine(runtime.NewInterpreterRuntime(runtime.Config{}))
+	ctx := fvm.NewContext(zerolog.Nop())
+	emptyPrograms := programs.NewEmptyPrograms()
+
+	view := testutil.RootBootstrappedLedger(vm, ctx)
+	v := view.NewChild()
+
+	sth := state.NewStateTransaction(
+		v,
+		state.DefaultParameters(),
+	)
+
+	return fvm.NewScriptEnvironment(context.Background(), ctx, vm, sth, emptyPrograms)
+}
+
+func (r *TestRunner) parseAndCheckImport(location common.Location, startCtx runtime.Context) (*ast.Program, *sema.Elaboration, error) {
+	if r.importResolver == nil {
+		return nil, nil, ImportResolverNotProvidedError{}
+	}
+
+	code, err := r.importResolver(location)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Create a new (child) context, with new environment.
+
+	env := runtime.NewBaseInterpreterEnvironment(runtime.Config{})
+
+	ctx := runtime.Context{
+		Interface:   startCtx.Interface,
+		Location:    location,
+		Environment: env,
+	}
+
+	env.CheckerConfig.ImportHandler = func(
+		checker *sema.Checker,
+		importedLocation common.Location,
+		importRange ast.Range,
+	) (sema.Import, error) {
+		return nil, fmt.Errorf("nested imports are not supported")
+	}
+
+	env.CheckerConfig.ContractVariableHandler = contractVariableHandler
+
+	program, err := r.testRuntime.ParseAndCheckProgram([]byte(code), ctx)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return program.Program, program.Elaboration, nil
+}
+
+// PrettyPrintResults is a utility function to pretty print the test results.
+//
+func PrettyPrintResults(results Results) string {
+	var sb strings.Builder
+	sb.WriteString("Test results:\n")
+	for _, result := range results {
+		sb.WriteString(PrettyPrintResult(result.TestName, result.Error))
+		sb.WriteRune('\n')
+	}
+	return sb.String()
+}
+
+func PrettyPrintResult(funcName string, err error) string {
+	if err == nil {
+		return fmt.Sprintf("- PASS: %s", funcName)
+	}
+
+	// Indent the error messages
+	errString := strings.ReplaceAll(err.Error(), "\n", "\n\t\t\t")
+
+	return fmt.Sprintf("- FAIL: %s\n\t\t%s", funcName, errString)
+}
```

Comment:
```suggestion
func PrettyPrintResult(funcName string, err error) string {
	if err != nil {
		// Indent the error messages
		errString := strings.ReplaceAll(err.Error(), "\n", "\n\t\t\t")

		return fmt.Sprintf("- FAIL: %s\n\t\t%s", funcName, errString)
	}
	
	return fmt.Sprintf("- PASS: %s", funcName)
}
```

indent the happy path to the left

Reply:
For the pretty printer, both are sort of happy paths (printing output - passed tests vs failed tests). So I put the path with 'less work' inside the condition. i.e: early exit. So less code to indent - easy on eyes.